community/dbms/src/main/java/org/neo4j/commandline/dbms/StoreLockChecker.java

```diff
@@ -58,10 +59,24 @@ static Closeable check( Path databaseDirectory ) throws CannotWriteException
         {
             if ( Files.isWritable( lockFile ) )
             {
-                StoreLockChecker storeLocker = new StoreLockChecker( new DefaultFileSystemAbstraction(),
-                        databaseDirectory.toFile() );
-                storeLocker.checkLock();
-                return storeLocker;
+                StoreLockChecker storeLocker = new StoreLockChecker( new DefaultFileSystemAbstraction(), databaseDirectory.toFile() );
+                try
+                {
+                    storeLocker.checkLock();
+                    return storeLocker;
+                }
+                catch ( StoreLockException le )
+                {
+                    try
+                    {
+                        storeLocker.close();
```

Comment:
We've seen this before from previous attempts to "clean up" around this area, at least three times. We can't seem to do this because then the original lock gets released.